tests/kafkatest/tests/produce_consume_validate.py

```diff
@@ -80,14 +98,16 @@ def run_produce_consume_validate(self, core_test_action=None, *args):
 
     @staticmethod
     def annotate_missing_msgs(missing, acked, consumed, msg):
-        msg += "%s acked message did not make it to the Consumer. They are: " % len(missing)
-        if len(missing) < 20:
-            msg += str(missing) + ". "
+        missing_list = list(missing)
+        msg += "%s acked message did not make it to the Consumer. They are: " %\
+            len(missing_list)
+        if len(missing_list) < 20:
+            msg += str(missing_list) + ". "
         else:
             for i in range(20):
-                msg += str(missing.pop()) + ", "
+                msg += str(missing_list[i]) + ", "
```

Comment:
Actually, the right way to do the join is `", ".join(str(m) for m in missing_list[:20])`, since `missing_list` is a list of ints, and `join` expects strings. Updated the PR with this fix.